src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -40,14 +40,14 @@ public function __construct(array $headers = array())
      */
     public function __toString()
     {
-        if (!$this->headers) {
+        if (!$headers = $this->all()) {
```

Comment:
If you remove `headers` usage, the property should be removed as well... but it is protected, so that's a BC break.